### ISSUE TITLE
[Public project]Re-push a deleted Chart via Helm V3 fails by 401

### ISSUE THREAD

**Issue by wy65701436** [goharbor/harbor#11271]
- push one chart with helm v3
- delete it in harbor
- Re-push it

After that, I get a 401. 

```
The push refers to repository [10.193.21.1/library/dummy-chart]
ref:     10.193.21.1/library/dummy-chart:0.1.0
digest:  3e94f4f36ce87478ab22cbb459aa2f93f45b0f0635407cd81cb5262d7c509a3b
size:    3.2 KiB
name:    dummy-chart
version: 0.1.0
Error: failed commit on ref "manifest-sha256:a8c04b6060af0ac7bfa315203ba917371808adfc57630014563efe85f3bd56b2": unexpected status: 401 Unauthorized
```

The root cause is probably that Harbor returns 200 to helm for the manifest head request, and helm client doesn't take auth header in the following PUT request.

Same thing happens on `docker app push`

**Comment by reasonerjt** [comment]
This is related to #11266 

Current response in `/v2` breaks some existing CLI tools when they push artifact to public projects.

**Comment by reasonerjt** [comment]
Fixed in #11406

Verified in build ` v2.0.0-42801b76`
